GAME/CONTROL.C

```diff
@@ -1045,9 +1046,35 @@ void RemoveRoomFlipItems(struct room_info* r)//1F938(<), 1FB4C(<) (F)
 	}
 }
 
-void FlipMap(int FlipNumber)
+void FlipMap(int FlipNumber) // (F)
 {
-	S_Warn("[FlipMap] - Unimplemented!\n");
+	struct room_info* r = room;
+	for(int i = 0; i < number_rooms; i++, r++)
+	{
+		if (r->flipped_room >= 0 && r->FlipNumber == FlipNumber)
+		{
+			struct room_info temp;
+			struct room_info* flipped = &room[r->flipped_room];
+			RemoveRoomFlipItems(r);
+			qmemcpy(&temp, r, sizeof(struct room_info));
+			qmemcpy(r, flipped, sizeof(struct room_info));
+			qmemcpy(flipped, &temp, sizeof(struct room_info));
+			r->flipped_room = flipped->flipped_room;
+			flipped->flipped_room = -1;
+			r->item_number = flipped->item_number;
+			r->fx_number = flipped->fx_number;
+			AddRoomFlipItems(r);
+		}
+	}
+	flip_stats[FlipNumber] = flip_stats[FlipNumber] == 0;
+	flip_status = flip_stats[FlipNumber] == 0;
+	{
+		struct creature_info* cinfo = baddie_slots;
```

Comment:
Must be declared at the very beginning of method.

Reply:
c89 only wants at the beginning of scope (which is why I used the { } brackets here, to create a new block scope)